python/ray/ml/trainer.py

```diff
@@ -54,6 +58,21 @@ def __init__(
 
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def training_function(
```

Comment:
```suggestion
    def training_loop(
```

Let's avoid undescriptive terms like function: everything's a function.

Reply:
Changed